src/super_gradients/module_interfaces/module_interfaces.py

```diff
@@ -69,3 +70,29 @@ def replace_num_classes(self, num_classes: int, compute_new_weights_fn: Callable
         :return: None
         """
         raise NotImplementedError(f"replace_num_classes is not implemented in the derived class {self.__class__.__name__}")
+
+
+class SupportsReplaceInChannels(ABC):
```

Comment:
Maybe use full name: In -> Input

Reply:
[3be5c64](https://github.com/Deci-AI/super-gradients/pull/1557/commits/3be5c64ac670dc5a27e8383193acaeb8272e3aa8)

Reply:
I hesitate `SupportsReplaceInputChannels` -> `SupportsInputChannels` or something like that, because I added `get_input_channels` to this abstract mixin, so it's not only "replace".
But at the same time, `SupportsInputChannels` is a very weird name. So maybe keeping `SupportsReplaceInputChannels` is better, `get_input_channels` being implicit in the name, but still explicit in the abstract class definition.

Reply:
`SupportsReplaceInputChannels` sounds absolutely fine to my taste (Much better than SupportsInputChannels).
And having get_input_channels is part of mixin is also fine since it's part of the contract one must implement to support this mixin. So the design is great here, IMO.